internal/consul/consul.go

```diff
@@ -11,6 +11,8 @@ type Client interface {
 	Catalog() Catalog
 	Health() Health
 	KV() KV
+	LockTrento(prefix string) (*consulApi.Lock, error)
```

Comment:
I named it `LockTrento` because Consul already has the `Lock` function, and I didn't want to create any confusion later on anyone trying to check the Consul `Lock` function signature and functionality.

We can name it `LockCustom` or any other name, but `Lock` might be risky

Reply:
We could use something like `LockKV` or `LockKey` or anything a bit more specific to avoid confusion

Reply:
I agree with that, locks should be as specific as possible. do we need a global lock on any manipulation of a KV entry? or only under certain paths? the name should reflect how specific the lock is imho

Reply:
@arbulu89 well, this is what go packages are for: the entire name of this symbol would be `consul.Lock()`, so I don't see any room for confusion. Otherwise, we would have had to use this `Trento` suffix for every other method of this interface, which we can agree would be really ugly!

Reply:
`LockKey(key string)` is the upstream API signature we are wrapping around, so yes, I would probably match that exactly.

Reply:
Well, the real consul `LockKey` is a totally different thing (it just returns an not use `Lock` struct), that's why I chose other name for this method, I didn't want people reading this `LockKey`, to think that it is the "official" consul `LockKey`. My new implementtion lock performs the lock itself. So it is not a wrapping itself as the other methods.
If you all prefer `LockKey` let's go for it, but this implements a totally different thing that the official consul `LockKey` method

Reply:
Okay, so while upstream only creates the handle for the lock, and one then has to call `lock()` on the lock handle to acquire it, our method creates _and_ acquires the lock in one go. Is that right?

Then our method could be called `AcquireKeyLock(key string)`. Does that work for everyone?

Reply:
Nice, I like it. But I would put it as `AcquireLockKey(key string)`

Reply:
One alternative is that you mimic the exact same behaviour of the upstream method, and we don't immediately acquire the lock, we just return the lock handle configured with our own custom session parameters and then explicitly call lock on it on the consumer side.
If that's not too much hassle for you, I would actually prefer this, then we can actually call it `LockKey` and mask the upstream method, because they become basically the same thing, only ours has some different pre-set from the Consul internal locks.

Reply:
Well, I just created this function to reduce the code in the consumers, and remove duplicates. It makes UT easier (having only in one spot, otherwise, we need to do always tricky things to mock the created `lock` instance.

Besides this, I don't really feel comfortable changing the consul api code calls. If for some reason somebody would need to use the original method, it wouldn't be possible.

Let's go with `AcquireLockKey`. It is not that important decision, and it can be reworked easily in the future if we find some need.

Reply:
All right then!